acceleration-everest-03-dotnet/Api/Program.cs

```diff
@@ -17,15 +17,17 @@
 var connectionString = builder.Configuration.GetConnectionString("DefaultConnection");
 builder.Services.AddDbContext<WarrenEverestDotnetDbContext>(
     dbContextOptions => dbContextOptions
-        .UseMySql(connectionString, ServerVersion.AutoDetect(connectionString)));
+        .UseMySql(connectionString, ServerVersion.AutoDetect(connectionString)), ServiceLifetime.Transient);
 builder.Services.AddControllers();
 builder.Services.AddSwaggerGen();
+builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddFluentValidationAutoValidation();
 builder.Services.AddTransient<ICustomerService, CustomerService>();
 builder.Services.AddTransient<ICustomerAppService, CustomerAppService>();
-builder.Services.AddValidatorsFromAssembly(Assembly.Load(nameof(AppServices)));
 builder.Services.AddAutoMapper(Assembly.Load(nameof(AppServices)));
+builder.Services.AddUnitOfWork<WarrenEverestDotnetDbContext>();
```

Comment:
Se o banco foi configurado para ciclo de vida Transient, devemos fazer o mesmo para o UnitOfWork
```suggestion
builder.Services.AddUnitOfWork<WarrenEverestDotnetDbContext>(ServiceLifetime.Transient);
```